Jobs are failing to render on "My Jobs" page

some jobs in "My Jobs" page can't be shown

![Image](https://github.com/user-attachments/assets/f7f5f3bf-6eff-4d01-9f03-4deca962ab74)


console:
```
ProfileJob.svelte:36 TypeError: t.arguments is not iterable
    at ProfileJob.svelte:23:58
    at async ProfileJob.svelte:41:9
```

JobId: 2ece3cba-a326-4877-85b1-9cc083a8c83d
JobId: 22b73bce-9415-4c33-b5a4-21ec30a0d19e

```
b99b-4c8e-ae73-2d221b568112/junit/get_all => generated 30 bytes in 7 msecs (HTTP/1.1 200) 3 headers in 85 bytes (1 switches on core 0)
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]: [ERROR] <5.29.124.170 - https://argus.scylladb.com/api/v1/test-info?testId=22b73bce-9415-4c33-b5a4-21ec30a0d19e - api.test_info> - error_handlers::handle_api_exception - Exception in api.test_info
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]: Traceback (most recent call last):
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/.cache/pypoetry/virtualenvs/argus-alm-HqUqk8xE-py3.12/lib/python3.12/site-packages/cassandra/cqlengine/query.py", line 775, in get
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     obj = self[0]
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:           ~~~~^^^
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/.cache/pypoetry/virtualenvs/argus-alm-HqUqk8xE-py3.12/lib/python3.12/site-packages/cassandra/cqlengine/query.py", line 573, in __getitem__
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     return self._result_cache[s]
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:            ~~~~~~~~~~~~~~~~~~^^^
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]: IndexError: list index out of range
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]: During handling of the above exception, another exception occurred:
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]: Traceback (most recent call last):
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/.cache/pypoetry/virtualenvs/argus-alm-HqUqk8xE-py3.12/lib/python3.12/site-packages/flask/app.py", line 880, in full_dispatch_request
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     rv = self.dispatch_request()
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:          ^^^^^^^^^^^^^^^^^^^^^^^
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/.cache/pypoetry/virtualenvs/argus-alm-HqUqk8xE-py3.12/lib/python3.12/site-packages/flask/app.py", line 865, in dispatch_request
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/app/argus/backend/service/user.py", line 261, in wrapped_view
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     return view(*args, **kwargs)
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:            ^^^^^^^^^^^^^^^^^^^^^
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/app/argus/backend/controller/api.py", line 385, in test_info
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     info = service.get_test_info(test_id=UUID(test_id))
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/app/argus/backend/service/argus_service.py", line 156, in get_test_info
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     test = ArgusTest.get(id=test_id)
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:            ^^^^^^^^^^^^^^^^^^^^^^^^^
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/.cache/pypoetry/virtualenvs/argus-alm-HqUqk8xE-py3.12/lib/python3.12/site-packages/cassandra/cqlengine/models.py", line 700, in get
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     return cls.objects.get(*args, **kwargs)
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/.cache/pypoetry/virtualenvs/argus-alm-HqUqk8xE-py3.12/lib/python3.12/site-packages/cassandra/cqlengine/query.py", line 763, in get
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     return self.filter(*args, **kwargs).get()
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:   File "/home/argus/.cache/pypoetry/virtualenvs/argus-alm-HqUqk8xE-py3.12/lib/python3.12/site-packages/cassandra/cqlengine/query.py", line 777, in get
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]:     raise self.model.DoesNotExist
Nov 20 10:26:38 ip-10-0-2-112 bash[875345]: cassandra.cqlengine.models.DoesNotExist
```

first sign of trouble, is at 15:05:

```
Nov 19 15:05:33 ip-10-0-2-112 bash[847502]: [ERROR] <87.71.200.23 - https://argus.scylladb.com/api/v1/test/22b73bce-9415-4c33-b5a4-21ec30a0d19e/runs?limit=10 - api.testrun_api.get_ru
ns_for_test> - error_handlers::handle_api_exception - Exception in api.testrun_api.get_runs_for_test
Nov 19 15:05:33 ip-10-0-2-112 bash[847502]: [pid: 847502|app: 0|req: 4215/283616] 87.71.200.23 () {58 vars in 3584 bytes} [Tue Nov 19 15:05:32 2024] GET /api/v1/test/22b73bce-9415-4c33-b5a4-21ec30a0d19e/runs?limit=10 => generated 74 bytes in 28 msecs (HTTP/1.1 200) 3 headers in 85 bytes (1 switches on core 1)
Nov 19 15:06:00 ip-10-0-2-112 bash[847501]: [ERROR] <93.173.89.28 - https://argus.scylladb.com/api/v1/test/22b73bce-9415-4c33-b5a4-21ec30a0d19e/runs?additionalRuns[]=fb7d3111-9329-4e0e-a237-715b6874a7ef&additionalRuns[]=f28b76ff-626d-43a5-9250-41452e831d06&limit=10 - api.testrun_api.get_runs_for_test> - error_handlers::handle_api_exception - Exception in api.testrun_api.get_runs_for_test
```

--- a/argus/backend/controller/main.py
+++ b/argus/backend/controller/main.py
@@ -44,6 +44,9 @@ def runs(test_id: UUID):
 @login_required
 def get_run_by_plugin(plugin_name: str, run_id: UUID | str):
     run = TestRunService().get_run(plugin_name, run_id)
+    if not run:
+        flash(f"Run {plugin_name}/{run_id} not found.", "error")
+        return redirect(url_for("main.error", type=404))
     return render_template("run_view_by_plugin.html.j2", run=run)
 
 

--- a/frontend/Profile/ProfileJob.svelte
+++ b/frontend/Profile/ProfileJob.svelte
@@ -3,7 +3,7 @@
     import { fade } from "svelte/transition";
     import { sendMessage } from "../Stores/AlertStore";
     import { extractBuildNumber } from "../Common/RunUtils";
-    import { InvestigationBackgroundCSSClassMap, InvestigationStatusIcon, StatusBackgroundCSSClassMap } from "../Common/TestStatus";
+    import { InvestigationStatusIcon, StatusBackgroundCSSClassMap } from "../Common/TestStatus";
     import { faChevronCircleDown, faChevronCircleUp } from "@fortawesome/free-solid-svg-icons";
     import Fa from "svelte-fa";
     import TestRuns from "../WorkArea/TestRuns.svelte";
@@ -12,6 +12,7 @@
     export let jobs = [];
     const dispatch = createEventDispatcher();
     let testInfo;
+    let failedFetch = false;
     let investigating = false;
 
     const fetchTestInfo = async function () {
@@ -33,6 +34,7 @@
             } else {
                 sendMessage("error", "Error fetching user jobs\nUnknown error. Check console for details", "ProfileJobs::testInfo");
             }
+            failedFetch = true;
             console.log(e);
         }
     };
@@ -87,9 +89,16 @@
             {/if}
         </div>
     {:else}
-        JobId: {jobId} <span class="spinner-grow"></span> Loading...
-        <div>
-            Total Jobs: {jobs.length}
-        </div>
+        {#if failedFetch}
+            <div>
+                Unable to load test info. The test might have been removed.
+                <div>TestId: {jobId}</div>
+            </div>
+        {:else}
+            TestId: {jobId} <div><span class="spinner-grow"></span> Loading...</div>
+            <div>
+                Total Jobs: {jobs.length}
+            </div>
+        {/if}
     {/if}
 </div>

